src/tools/tidy/src/main.rs

```diff
@@ -31,7 +31,7 @@ fn main() {
 
     // Checks that only make sense for the compiler.
     errors::check(&compiler_path, &mut bad);
-    error_codes_check::check(&src_path, &mut bad);
+    error_codes_check::check(&[&src_path, &compiler_path], &mut bad);
```

Comment:
Hm, what error codes exist under src/? Should we actually split this into two separate passes, one to collect tests under src/test/ui/error-codes and the other for errors in the compiler?

Reply:
Under `src` are the tests. We check that all error codes are emitted at least in one test. I think it'd be hard to split the two parts considering that on one side we iterate over all emitted error codes and on the other we list all error codes. At the end, we check that all listed error codes have been tested once.